### DISCUSSION TITLE
Why does Capacitor choose to run the app on a localhost server instead of just from the filesystem.

### DISCUSSION THREAD

**Discussion by tafelnl** [ionic-team/capacitor#3211] [General] [answered]
I do not understand what advantages come from using the `http://localhost` instead of just simply `file:///`. Can someone explain that to me?

**Comment by jcesarmobile** [comment, marked as answer]
Using `file:` protocol to load the app code has a few bugs/problems, like html 5 routing not working, localStorage (or any web storage) can be lost on uri change, on iOS you can't fetch/XHR local files, etc.

All those problems go away if using http/https on Android and a custom scheme on iOS.

**Reply by IT-MikeS** [reply]
Taking the same approach for similar issues in electron community platform
